### PR TITLE
Destructuring assignment

### PR DESCRIPTION
**Pull request by Arash-1999**

### CODE REVIEW

File: 1-js/05-data-types/10-destructuring-assignment/1-destruct-user/task.md [hunk 1]
```diff
@@ -13,18 +13,18 @@ let user = {
 };
 ```
 
-Write the destructuring assignment that reads:
+یک تخصیص ساختارشکنانه بنویسید که این شرایطرا داشته باشد:
```
Comment (kooshan75) [inline 1-js/05-data-types/10-destructuring-assignment/1-destruct-user/task.md:16]: Please put a whitespace between "شرایط" and "را".

File: 1-js/05-data-types/10-destructuring-assignment/article.md [hunk 2]
```diff
@@ -1,552 +1,532 @@
...
+# تخصیص ساختارشکنانه
 
-The two most used data structures in JavaScript are `Object` and `Array`.
+اشیا(ابجکت)و آرایه ها پرمصرفترین ساختار داده ها هستند.
 
-Objects allow us to create a single entity that stores data items by key, and arrays allow us to gather data items into an ordered collection.
+اشیا به ما اجازه میدهند که یک موجودیت که داده را با یک کلید ذخیره میکند٬ و آرایه ها به ما اجازه میدهند که داده ها را در یک مجموعه شماره گذاری شده ذخیره کنیم.
 
-But when we pass those to a function, it may need not an object/array as a whole, but rather individual pieces.
+اما زمانیکه ما آنها ره به یک تابع میدهیم٬ ممکن است یک شی یا آرایه را بطور کامل نخواهد٬ بلکه قسمت های جداگانه ای را بخواهد.
```
Comment (kooshan75) [inline 1-js/05-data-types/10-destructuring-assignment/article.md:7]: Change "ره" to "را"

File: 1-js/05-data-types/10-destructuring-assignment/article.md [hunk 3]
```diff
@@ -1,552 +1,532 @@
...
-// sets firstName = arr[0]
-// and surname = arr[1]
+// تخصیص ساختارشکنانه
+// متغیر اول را به عضو اول آرایه اختصاص میدهد
+// و دومی را به دومی
 let [firstName, surname] = arr;
 */!*
 
 alert(firstName); // Ilya
-alert(surname);  // Kantor
```
Comment (kooshan75) [inline 1-js/05-data-types/10-destructuring-assignment/article.md:27]: You don't need to change sample written code :)

File: 1-js/05-data-types/10-destructuring-assignment/article.md [hunk 4]
```diff
@@ -1,552 +1,532 @@
...
+حال میتوانی به جای کار کردن با آرایه از متغیرها استفاده کنیم.
 
-It looks great when combined with `split` or other array-returning methods:
+خیلی بهتر میشود که این را با `split` یا متد هایی که ارایه باز میگردانند ترکیب کنیم.
 
 ```js
 let [firstName, surname] = "Ilya Kantor".split(' ');
 ```
 
-````smart header="\"Destructuring\" does not mean \"destructive\"."
```
Comment (kooshan75) [inline 1-js/05-data-types/10-destructuring-assignment/article.md:38]: Also, this shouldn't be deleted as well.

File: 1-js/05-data-types/10-destructuring-assignment/article.md [hunk 5]
```diff
@@ -1,552 +1,532 @@
...
-It's called "destructuring assignment," because it "destructurizes" by copying items into variables. But the array itself is not modified.
+```
+به این تخصیص ساختارشکنانه میگویند٬ زیرا این ساختار را کپی کردن ایتم آیتم در متغیرها میشکند. ولی خود آرایه دست نخورده باقی میماند.
 
-It's just a shorter way to write:
+این فقط یک راه کوتاه تر برای نوشتن است.
 ```js
-// let [firstName, surname] = arr;
```
Comment (kooshan75) [inline 1-js/05-data-types/10-destructuring-assignment/article.md:43]: no need to delete the whitespace :)

File: 1-js/05-data-types/10-destructuring-assignment/article.md [hunk 6]
```diff
@@ -1,552 +1,532 @@
...
+//let [firstName, surname] = arr;
 let firstName = arr[0];
 let surname = arr[1];
 ```
-````
```
Comment (kooshan75) [inline 1-js/05-data-types/10-destructuring-assignment/article.md:47]: Don't delete the original backtick count.

File: 1-js/05-data-types/10-destructuring-assignment/article.md [hunk 7]
```diff
@@ -1,552 +1,532 @@
...
+```
 
-````smart header="Ignore elements using commas"
```
Comment (kooshan75) [inline 1-js/05-data-types/10-destructuring-assignment/article.md:49]: You've also deleted this.

File: 1-js/05-data-types/10-destructuring-assignment/article.md [hunk 8]
```diff
@@ -1,552 +1,532 @@
...
-Unwanted elements of the array can also be thrown away via an extra comma:
+```
+از المان هایی که نمیخواهیم میتوان با یک ویرگول اضافه چشمپوشی کرد.
 
 ```js run
 *!*
-// second element is not needed
-let [firstName, , title] = ["Julius", "Caesar", "Consul", "of the Roman Republic"];
```
Comment (kooshan75) [inline 1-js/05-data-types/10-destructuring-assignment/article.md:55]: Was this autocomplete? You don't need to change original sample code.

File: 1-js/05-data-types/10-destructuring-assignment/article.md [hunk 9]
```diff
@@ -1,552 +1,532 @@
...
+// نیازی به المان دوم نیست
+let [firstName, , title] = ["Julius", Cesar", "Counsul", "of the Rome Republic"];
 */!*
 
-alert( title ); // Consul
```
Comment (kooshan75) [inline 1-js/05-data-types/10-destructuring-assignment/article.md:58]: Keep the original code.

File: 1-js/05-data-types/10-destructuring-assignment/article.md [hunk 10]
```diff
@@ -1,552 +1,532 @@
...
+alert(title); //Counsul
+```
+در قطعه کد بالا٬‌ از المان دوم آرایه چشمپوشی شده و سومی به `title` اختصاص داده شده است. همچنین از بقیه آیتم ها چشمپوشی شده است چون متغیری برای آنها وجود ندارد.
 ```
 
-In the code above, the second element of the array is skipped, the third one is assigned to `title`, and the rest of the array items is also skipped (as there are no variables for them).
```
Comment (kooshan75) [inline 1-js/05-data-types/10-destructuring-assignment/article.md:61]: I think you've forgotten translating this.

File: 1-js/05-data-types/10-destructuring-assignment/article.md [hunk 11]
```diff
@@ -1,552 +1,532 @@
...
-````
-
-````smart header="Works with any iterable on the right-side"
```
Comment (kooshan75) [inline 1-js/05-data-types/10-destructuring-assignment/article.md:64]: Translation needed.

File: 1-js/05-data-types/10-destructuring-assignment/article.md [hunk 12]
```diff
@@ -1,552 +1,532 @@
...
-
-...Actually, we can use it with any iterable, not only arrays:
-
+```
+در واقع ما میتوانیم از هر چیز پیمودنی دیگر استفاده کنیم. نه فقط آرایه ها.
+```
 ```js
-let [a, b, c] = "abc"; // ["a", "b", "c"]
```
Comment (kooshan75) [inline 1-js/05-data-types/10-destructuring-assignment/article.md:69]: Keep the original code please.

**Review by kooshan75 — CHANGES_REQUESTED** [review]
There was too many changes in the original code that you need to redo and then resubmit again. Thanks for the contribution.

**Review by kooshan75 — CHANGES_REQUESTED** [review]
I think you didn't read my comments on your translation carefully. please read and fix them. Thanks.

**Comment by Arash-1999** [comment]
\done

**Comment by mehradsadeghi** [comment]
Dear @kooshan75 please `resolve` each comment that is being fixed by @Arash-1999.

**Comment by kooshan75** [comment]
Dear @Arash-1999 I reviewed your translation but there were too many code changes and conflicts that has made it too hard for me to even review the changes that have been made. I humbly ask to keep the original code and replace the translation parts only. I will close this PR and please submit another PR with resolved conflicts and code changes. Best of luck.

**Pull request closed without merging**